### PR TITLE
Adds welding hardhats

### PR DESCRIPTION
**Pull request by BuffEngineering**
:cl: BuffEngineering/MrDoomBringer
add: Engineers may now weld with engineering pride using the new welding hardhats!
/:cl:

Adds a hardhat that has both a light and a welding visor in the yellow, orange, white, and blue colors. The non-white variants are a bit dimmer than a normal hardhat to make them different from a hardhat and welding goggles. The CE has access to the white variant in his locker and the yellow variant is available in the EngiDrobe. Sprites have been updated, let me know if you have any suggestions.
weldhatnew

### CODE REVIEW

File: code/modules/clothing/head/hardhat.dm [hunk 1]
```diff
@@ -81,4 +81,90 @@
 	heat_protection = HEAD
 	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT
 	cold_protection = HEAD
-	min_cold_protection_temperature = FIRE_HELM_MIN_TEMP_PROTECT
\ No newline at end of file
+	min_cold_protection_temperature = FIRE_HELM_MIN_TEMP_PROTECT
+
+/obj/item/clothing/head/hardhat/weldhat/
```
Comment (Trilbyspaceclone) [inline code/modules/clothing/head/hardhat.dm]: /obj/item/clothing/head/hardhat/weldhat/ -> /obj/item/clothing/head/hardhat/weldhat

Comment (Trilbyspaceclone) [reply code/modules/clothing/head/hardhat.dm]: Just remove the spare / at the end and by the ***LOOKS*** it should pass ya goof

Comment (BuffEngineering) [reply code/modules/clothing/head/hardhat.dm]: Thanks

Comment (Trilbyspaceclone) [reply code/modules/clothing/head/hardhat.dm]: Looking at it, runtiming with white one?

Comment (BuffEngineering) [reply code/modules/clothing/head/hardhat.dm]: Thanks for pointing it out, I think I got it.

**Comment by subject217** [comment]
nice username

**Comment by tralezab** [comment]
Can we see the welding hardhats in game?

**Comment by BuffEngineering** [comment]
Figured I would forget something, thanks for the reminder.

File: code/modules/clothing/head/hardhat.dm [hunk 2]
```diff
@@ -81,4 +81,92 @@
...
+	visor_flags_inv = HIDEEYES 
+	visor_flags_cover = HEADCOVERSEYES
+
+/obj/item/clothing/head/hardhat/weldhat/Initialize()
+	..()
+	update_icon()
+
+/obj/item/clothing/head/hardhat/weldhat/attack_self(mob/user)
+	toggle_helmet_light(user)
+	update_icon()
```
Comment (Jordie0608) [inline code/modules/clothing/head/hardhat.dm]: You're already calling `update_icon()` at the end of `toggle_helmet_light()`, no need to call again.

File: code/modules/clothing/head/hardhat.dm [hunk 3]
```diff
@@ -81,4 +81,92 @@
...
+
+/obj/item/clothing/head/hardhat/weldhat/AltClick(mob/user)
+	toggle_welding_screen(user)
+	update_icon()
```
Comment (Jordie0608) [inline code/modules/clothing/head/hardhat.dm]: Same.

**Review by Jordie0608 — CHANGES_REQUESTED** [review]
These seem just better, is there any reason to have a hardhat when you could have a welding hardhat?

**Comment by BuffEngineering** [comment]
Thanks for pointing that redundancy out. The main difference is that the welding hard hat is dimmer by a tile, aside from the CE's variant, so it's a minor trade off in being able to see less dangers in general for the convenience of having a welding visor. It also lets engineers have the option to keep the engineering style without using googles that have a mad scientist feel to them.

**Comment by Tlaltecuhtli** [comment]
the frontal sprite looks bad tho

**Comment by MrDoomBringer** [comment]
@WarJenkins shit my b here u go
⠕⠓ ⠛⠕⠙ ⠕⠓ ⠋⠥⠉⠅⠕⠓ ⠛⠕⠙ ⠕⠓ ⠋⠥⠉⠅

**Comment by Tlaltecuhtli** [comment]
why no WIP tag if the sprites are temp reeeeeee

**Comment by MrDoomBringer** [comment]
@Tlaltecuhtli cus i was supposed to finish the sprites like an hour after the pr was posted kek

File: code/modules/clothing/head/hardhat.dm [hunk 4]
```diff
@@ -81,4 +81,90 @@
...
+	flash_protect = 2
+	tint = 2
+	flags_inv = HIDEEYES
+	flags_cover = HEADCOVERSEYES
+	visor_vars_to_toggle = VISOR_FLASHPROTECT | VISOR_TINT
+	visor_flags_inv = HIDEEYES 
+	visor_flags_cover = HEADCOVERSEYES
+
+/obj/item/clothing/head/hardhat/weldhat/Initialize()
+	..()
```
Comment (Cyberboss) [inline code/modules/clothing/head/hardhat.dm]:
```suggestion
	. = ..()
```

File: code/modules/clothing/head/hardhat.dm [hunk 5]
```diff
@@ -81,4 +81,90 @@
...
+	update_icon()
+
+/obj/item/clothing/head/hardhat/weldhat/attack_self(mob/user)
```
Comment (Cyberboss) [inline code/modules/clothing/head/hardhat.dm]:
```suggestion
/obj/item/clothing/head/hardhat/weldhat/attack_self(mob/living/user)
```

File: code/modules/clothing/head/hardhat.dm [hunk 6]
```diff
@@ -81,4 +81,90 @@
...
+/obj/item/clothing/head/hardhat/weldhat/proc/toggle_helmet_light(mob/user)
+	on = !on	
+	if(on)
+		turn_on(user)
+	else
+		turn_off(user)
+	update_icon()
+
+/obj/item/clothing/head/hardhat/weldhat/proc/toggle_welding_screen(mob/user)
+	playsound(src.loc, 'sound/mecha/mechmove03.ogg', 50, 1) //Visors don't just come from nothing
```
Comment (Cyberboss) [inline code/modules/clothing/head/hardhat.dm]:
```suggestion
	playsound(src, 'sound/mecha/mechmove03.ogg', 50, 1) //Visors don't just come from nothing
```

File: code/modules/clothing/head/hardhat.dm [hunk 7]
```diff
@@ -81,4 +81,90 @@
...
+/obj/item/clothing/head/hardhat/weldhat/AltClick(mob/user)
+	toggle_welding_screen(user)
+
+/obj/item/clothing/head/hardhat/weldhat/ui_action_click(mob/user, action)
+	if(istype(action, /datum/action/item_action/toggle_helmet_light))
+		toggle_helmet_light(user)
+	else
+		toggle_welding_screen(user)
+
+/obj/item/clothing/head/hardhat/weldhat/proc/toggle_helmet_light(mob/user)
```
Comment (Cyberboss) [inline code/modules/clothing/head/hardhat.dm]: This proc contains duplicated code from `/obj/item/clothing/head/hardhat/attack_self`

File: code/modules/clothing/head/hardhat.dm [hunk 8]
```diff
@@ -81,4 +81,90 @@
...
+
+/obj/item/clothing/head/hardhat/weldhat/Initialize()
+	..()
+	update_icon()
+
+/obj/item/clothing/head/hardhat/weldhat/attack_self(mob/user)
+	toggle_helmet_light(user)
+
+/obj/item/clothing/head/hardhat/weldhat/AltClick(mob/user)
+	toggle_welding_screen(user)
```
Comment (Cyberboss) [inline code/modules/clothing/head/hardhat.dm]: Needs ghost checks

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by BuffEngineering** [comment]
The requested changes are complete, thanks for pointing those out. Also new sprites.

**Comment by Tlaltecuhtli** [comment]
i dont get the point tho, engineering starts with welding hats and welding goggles and after like 5 min they get welding gasmask from techwebs(tm), flashlights, pda lights and hardhat for light it seems redunant to add an hat that is a direct upgrade to all you have roundstart and is also free/roundstart

**Review by Jordie0608 — DISMISSED** [review]

**Comment by MrDoomBringer** [comment]
@Tlaltecuhtli LOTS of departments get ""straight upgrades"" to standard crew gear specifically because it's something that falls within their specialization in order to make doing their job easier than some random greyshit doing it.

Thats like complaining that botany gets free high capacity water tanks (as opposed to the crew's regular tanks), or that security gets free stunbatons (as opposed to the crew's slow+bulky stunprods), or that atmos gets a free fireproof hardsuit (as opposed to the crew's normal spacesuits), or that medbay gets free styptic spray (as opposed to the crew's generic brute patches)

**Comment by Tlaltecuhtli** [comment]
im not talking about upgrade over standard grey shirt (now that public autolathe is gone they start with 0 welder protection except maint sunglasses) but it seems weird that you can choose between hard hat or better hard hat on the same vendor and both free, it should be a premium item at least

File: code/modules/clothing/head/hardhat.dm [hunk 9]
```diff
@@ -14,19 +14,27 @@
...
 
-/obj/item/clothing/head/hardhat/attack_self(mob/user)
-	on = !on
-	icon_state = "hardhat[on]_[item_color]"
-	item_state = "hardhat[on]_[item_color]"
-	user.update_inv_head()	//so our mob-overlays update
+/obj/item/clothing/head/hardhat/attack_self(mob/living/user)
+	toggle_helmet_light(user)
 
+/obj/item/clothing/head/hardhat/proc/toggle_helmet_light(mob/user)
```
Comment (Cyberboss) [inline code/modules/clothing/head/hardhat.dm]:
```suggestion
/obj/item/clothing/head/hardhat/proc/toggle_helmet_light(mob/living/user)
```

File: code/modules/clothing/head/hardhat.dm [hunk 10]
```diff
@@ -81,4 +89,75 @@
...
+	update_icon()
+
+/obj/item/clothing/head/hardhat/weldhat/attack_self(mob/living/user)
+	toggle_helmet_light(user)
+
+/obj/item/clothing/head/hardhat/weldhat/AltClick(mob/user)
+	if(user.canUseTopic(src, BE_CLOSE))
+		toggle_welding_screen(user)
+
+/obj/item/clothing/head/hardhat/weldhat/ui_action_click(mob/user, action)
```
Comment (Cyberboss) [inline code/modules/clothing/head/hardhat.dm]: Do not use ui_action_click, make a custom action instead

File: code/modules/clothing/head/hardhat.dm [hunk 11]
```diff
@@ -81,4 +89,75 @@
...
+	if(istype(action, /datum/action/item_action/toggle_helmet_light))
+		toggle_helmet_light(user)
+	else
+		toggle_welding_screen(user)
+
+/obj/item/clothing/head/hardhat/weldhat/proc/toggle_welding_screen(mob/user)
```
Comment (Cyberboss) [inline code/modules/clothing/head/hardhat.dm]:
```suggestion
/obj/item/clothing/head/hardhat/weldhat/proc/toggle_welding_screen(mob/living/user)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by FantasticFwoosh** [comment]
I preferred the original concept sprite, it would have justified making one out of (Plastinium Glass x1, wire x1, welder x1, screwdriver x1, hardhat x1) like the sunglasses / sunglasses huds with a deconstruction recipie if the power of it is really as weak as the OP describes with the similar reduced viewchange of a regular welding helmet.

- Now its just clear, you can't tell whether it's a faceguard or not.

I wouldn't be against a aesthetic change either for welding gas-masks this is supposed to early-game make redundant , but the function is intact.

**Pull request merged**